ideascube/mediacenter/tests/test_clean.py

```diff
@@ -0,0 +1,141 @@
+    original_path = document.original.path
+    preview_path = document.preview.path
+    document.delete()
+    assert os.path.exists(original_path)
+    assert os.path.exists(preview_path)
+    os.chmod(os.path.join(settings.MEDIA_ROOT, 'mediacenter/document'), stat.S_IRUSR | stat.S_IXUSR)
+    call_command('clean', 'leftover-files')
+    assert os.path.exists(original_path)
+    assert not os.path.exists(preview_path)
+    _, err = capsys.readouterr()
+    assert "ERROR while deleting {}".format(original_path) in err
+
+
+def test_clean_leftover_should_not_remove_ophan_files_if_dry_run(settings, capsys):
+    document = DocumentFactory(original__from_path=os.path.join(DATA_PATH, 'a-video.mp4'),
+                               preview__from_path=os.path.join(DATA_PATH, 'an-image.jpg'))
+    original_path = document.original.path
+    preview_path = document.preview.path
+    Document.objects.all().delete()
+    assert os.path.exists(original_path)
+    assert os.path.exists(preview_path)
+    call_command('clean', 'leftover-files', dry_run=True)
+    assert os.path.exists(original_path)
+    assert os.path.exists(preview_path)
+    out, _ = capsys.readouterr()
+    assert original_path in out
+    assert preview_path in out
+
+
+def test_clean_leftover_should_correctly_clean_after_document_update(settings):
+    from django.core.files import File
+    document = DocumentFactory(original__from_path=os.path.join(DATA_PATH, 'a-video.mp4'),
+                               preview__from_path=os.path.join(DATA_PATH, 'an-image.jpg'))
+    old_original_path = document.original.path
+    preview_path = document.preview.path
+    with open(os.path.join(DATA_PATH, 'a-pdf.pdf'), 'rb') as f:
+        document.original = File(f, name='a-pdf.pdf')
+        document.save()
+    new_original_path = document.original.path
+    assert os.path.exists(old_original_path)
+    assert os.path.exists(new_original_path)
+    assert os.path.exists(preview_path)
+    call_command('clean', 'leftover-files')
+    assert not os.path.exists(old_original_path)
+    assert os.path.exists(new_original_path)
+    assert os.path.exists(preview_path)
+
+
+def test_clean_leftover_should_correctly_clean_after_complex_situation(settings):
+    from django.core.files import File
+    document0 = DocumentFactory(original__from_path=os.path.join(DATA_PATH, 'a-video.mp4'),
+                                preview__from_path=os.path.join(DATA_PATH, 'an-image.jpg'))
+    document1 = DocumentFactory(original__from_path=os.path.join(DATA_PATH, 'a-video.mp4'),
+                                preview__from_path=os.path.join(DATA_PATH, 'an-image.jpg'))
+    document2 = DocumentFactory(original__from_path=os.path.join(DATA_PATH, 'a-video.mp4'),
+                                preview=None)
+    document3 = DocumentFactory(original__from_path=os.path.join(DATA_PATH, 'a-video.mp4'),
+                                preview__from_path=os.path.join(DATA_PATH, 'an-image.jpg'))
```

Comment:
Oh, those paths are "before the upload".

You got me for a moment here, I thought the paths in database would be identical, which then made me wonder why the paths from `document1` would end up being deleted, since `document0` and document2` would still reference them.

In fact, that's one more good argument in favour of having it done this way rather than in the ORM/forms: we could have removed the path of a document when deleting another document if the two documents shared a path!